File: 4391-318-367-valid-perfect-square/4391-318-367-valid-perfect-square.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        if(num==1 || num==4)
          return true;
        for(int p=1, i=1,j=num/2;i<=j;p++)
          {   int mid = i + (j - i) / 2;
             long square = (long) mid * mid;
              if(square==num)
                return true;
              else if(square>num)
                j=mid-1;
              else
                i=mid+1;

          }
          return false;
    }
};

File: 4391-318-367-valid-perfect-square/4391-318-367-valid-perfect-square.py
class Solution(object):
    def isPerfectSquare(self, num):
        if num==1 or num==4:
            return True
        i=1
        j=num/2
        while i<=j:
            mid=(i+j)/2
            if mid*mid==num:
                return True
            elif mid*mid>num:
                j=mid-1
            else:
                i=mid+1
        return False
        """
        :type num: int
        :rtype: bool
        """
        